src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5135,6 +5135,19 @@ int RGWRados::delete_bucket(RGWBucketInfo& bucket_info, RGWObjVersionTracker& ob
     (void) CLSRGWIssueBucketIndexClean(index_pool.ioctx(),
 				       bucket_objs,
 				       cct->_conf->rgw_bucket_index_max_aio)();
+  } else {
+    bucket_info.flags |= BUCKET_DELETED;
+    map<string, bufferlist> attrs;
+
+    r = ctl.bucket->store_bucket_instance_info(bucket, bucket_info, y, dpp, RGWBucketCtl::BucketInstance::PutParams()
+                                                                    .set_exclusive(false)
+                                                                    .set_mtime(real_time())
+                                                                    .set_attrs(&attrs)
+                                                                    .set_orig_info(&bucket_info));
```

Comment:
these metadata writes are protected by cls_version, so we may get `ECANCELED` if there are racing writes. we need a retry loop here that re-reads the info and adds back the `BUCKET_DELETED` flag. same goes for the `store_bucket_instance_info()` call that adds the 'deleted' log layout

`set_attrs()` with empty attrs will delete any existing attrs. if we don't have access to the real attrs, we'll need to modify callers to pass them in

Reply:
got it

Reply:
done